src/common/ceph_time.cc

```diff
@@ -0,0 +1,112 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.	See file COPYING.
+ *
+ */
+
+// For ceph_timespec
+#include "include/types.h"
+
+#include "ceph_context.h"
+#include "ceph_time.h"
+#include "config.h"
+
+namespace ceph {
+  namespace time_detail {
+    real_clock::time_point real_clock::now(const CephContext* cct) noexcept {
+      struct timespec ts;
+      clock_gettime(CLOCK_REALTIME, &ts);
+      // TODO: After we get the time-literal configuration patch in,
+      // just add the configured duration.
+      if (cct)
+	ts.tv_sec += cct->_conf->clock_offset;
+      return from_timespec(ts);
+    }
+
+    void real_clock::to_ceph_timespec(const time_point& t,
+				      struct ceph_timespec& ts) {
+      ts.tv_sec = to_time_t(t);
+      ts.tv_nsec = (t.time_since_epoch() % seconds(1)).count();
+    }
+    struct ceph_timespec real_clock::to_ceph_timespec(const time_point& t) {
+      struct ceph_timespec ts;
+      to_ceph_timespec(t, ts);
+      return ts;
+    }
+    real_clock::time_point real_clock::from_ceph_timespec(
+      const struct ceph_timespec& ts) {
+      return time_point(seconds(ts.tv_sec) + nanoseconds(ts.tv_nsec));
+    }
+
+    coarse_real_clock::time_point coarse_real_clock::now(
```

Comment:
That is correct. It's currently unused. I added it because there was some thought it might be useful at some point. If, for example, we were logging elapsed time rather than basing logic on it and not querying a time for any other reason, a clock that was faster but lower resolution might be nice.